tests/test_kernels.cc

```diff
@@ -35,6 +35,20 @@ namespace rascal {
   using multiple_ref_fixtures =
       boost::mpl::list<KernelFixture<DataSphericalInvariantsKernelFixture>>;
 
+  /**
+   * Tests if the wrong target_type is catched correctly.
+   */
+  BOOST_FIXTURE_TEST_CASE(kernel_target_type_test,
+                          KernelFixture<DataSphericalInvariantsKernelFixture>) {
+    auto kernel_hyper = this->ParentA::kernel_hypers.at(0);
+    kernel_hyper["target_type"] = "this_target_type_does_not_exist";
```

Comment:
Again, how about a case with `kernel_hyper["target_type"] = "structure"` to show that it fails?